File: internal/expr/literal.go
package expr

type Literal struct {
	Value any
}

func NewLiteral(value any) Expr {
	return &Literal{
		Value: value,
	}
}

func (l *Literal) Accept(v Visitor) any {
	return v.VisitLiteralExpr(l)
}

File: src/expr/literal.rs
use crate::token::TokenValue;

use super::{ExprVisitor, Visitor};

pub struct Literal {
    pub value: TokenValue,
}

impl Literal {
    pub fn new(value: TokenValue) -> Self {
        Literal { value }
    }
}

impl Visitor<&dyn ExprVisitor<()>> for Literal {
    fn accept(&self, visitor: &dyn ExprVisitor<()>) {
        visitor.visit_literal_expr(self);
    }
}
